src/panels/lovelace/editor/card-editor/hui-entity-picker.ts

```diff
@@ -0,0 +1,132 @@
+import memoize from "memoize-one";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  TemplateResult,
+} from "lit-element";
+
+import { computeStateName } from "../../../../common/entity/compute_state_name";
+import { computeDomain } from "../../../../common/entity/compute_domain";
+import { computeRTLDirection } from "../../../../common/util/compute_rtl";
+import { fireEvent } from "../../../../common/dom/fire_event";
+
+import type { HASSDomEvent } from "../../../../common/dom/fire_event";
+import type { HomeAssistant } from "../../../../types";
+import type {
+  DataTableRowData,
+  SelectionChangedEvent,
+} from "../../../../components/data-table/ha-data-table";
+
+import "../../../../components/data-table/ha-data-table";
+import "../../../../components/entity/state-badge";
+
+@customElement("hui-entity-picker")
+export class HuiEntityPicker extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  protected render(): TemplateResult {
+    return html`
+      <div class="container">
+        <ha-data-table
+          auto-height
+          selectable
+          .id=${"entity_id"}
+          .columns=${this._columns()}
+          .data=${this._allEntities(this.hass).map((entity) => {
```

Comment:
We are still mapping and creating a new object on every update of hass, add that the memoize too, so it just returns the data object.